curp/src/server/bg_tasks.rs

```diff
@@ -67,25 +81,37 @@ pub(super) async fn run_bg_tasks<
         Arc::clone(&state),
         Arc::clone(&timeout),
     ));
-    let bg_apply_handle = tokio::spawn(bg_apply(Arc::clone(&state), cmd_exe_tx, spec));
+    let bg_apply_handle = tokio::spawn(bg_apply(Arc::clone(&state), cmd_exe_tx));
     let bg_heartbeat_handle = tokio::spawn(bg_heartbeat(
         connects.clone(),
         Arc::clone(&state),
         Arc::clone(&timeout),
     ));
     let bg_get_sync_cmds_handle =
         tokio::spawn(bg_get_sync_cmds(Arc::clone(&state), sync_chan, ae_trigger));
-    let calibrate_handle = tokio::spawn(bg_leader_calibrates_followers(
-        connects,
-        state,
-        Arc::clone(&timeout),
-    ));
 
     // spawn cmd execute worker
-    let bg_exe_worker_handles: Vec<JoinHandle<_>> =
-        iter::repeat((cmd_exe_rx, Arc::new(cmd_executor)))
-            .take(N_EXECUTE_WORKERS)
-            .map(|(rx, ce)| tokio::spawn(execute_worker(rx, ce)))
+    let cmd_executor = Arc::new(cmd_executor);
+    let spec = state.read().spec();
+    let bg_exe_worker_handles: Vec<JoinHandle<_>> = iter::repeat((
+        cmd_exe_rx,
+        state.read().cmd_board(),
+        Arc::clone(&cmd_executor),
+    ))
+    .take(N_EXECUTE_WORKERS)
+    .map(|(rx, cmd_board, ce)| {
+        tokio::spawn(execute_worker(
+            rx,
+            Arc::clone(&cmd_board),
+            Arc::clone(&spec),
+            ce,
+        ))
+    })
+    .collect();
+    let bg_as_worker_handles: Vec<JoinHandle<_>> =
+        iter::repeat((cmd_as_rx, state.read().cmd_board(), cmd_executor))
+            .take(N_AFTER_SYNC_WORKERS)
+            .map(|(rx, cmd_board, ce)| tokio::spawn(after_sync_worker(rx, cmd_board, ce)))
             .collect();
```

Comment:
```suggestion
    let cmd_executor = Arc::new(cmd_executor);
    let (spec, cmd_board) = state.map_read(|state_r| (state_r.spec(), state_r.cmd_board()));
    let bg_exe_worker_handles: Vec<JoinHandle<_>> = (0..N_EXECUTE_WORKERS)
        .map(|_i| {
            tokio::spawn(execute_worker(
                cmd_exe_rx.clone(),
                Arc::clone(&cmd_board),
                Arc::clone(&spec),
                Arc::clone(&cmd_executor),
            ))
        })
        .collect();
    let bg_as_worker_handles: Vec<JoinHandle<_>> = (0..N_AFTER_SYNC_WORKERS)
        .map(|_i| {
            tokio::spawn(after_sync_worker(
                cmd_as_rx.clone(),
                Arc::clone(&cmd_board),
                Arc::clone(&cmd_executor),
            ))
        })
        .collect();
```

Reply:
Fixed. Moved all state read to the beginning of the function.